ui/org.eclipse.pde.ui/src/org/eclipse/pde/internal/ui/views/dependencies/DependenciesViewTreePage.java

```diff
@@ -70,7 +74,9 @@ public DependenciesViewTreePage(DependenciesView view, ITreeContentProvider cont
 
 	@Override
 	protected StructuredViewer createViewer(Composite parent) {
-		fTreeViewer = new TreeViewer(parent, SWT.MULTI | SWT.V_SCROLL | SWT.H_SCROLL);
+		fFilteredTreeViewer = new FilteredTree(parent, SWT.MULTI | SWT.V_SCROLL | SWT.H_SCROLL, new PatternFilter(),
+				true, true);
```

Comment:
Unfortunately the bundles in a target-platform can have cycles and when such cycle is encountered one gets a `StackOverflowException`.
I have tried to create a guard for that situation, but I'm not yet sure this is sufficient for all cases. In general we could also think about adding that to `PatternFilter` directly (maybe optionally).
```suggestion
		fFilteredTreeViewer = new FilteredTree(parent, SWT.MULTI | SWT.V_SCROLL | SWT.H_SCROLL, new PatternFilter() {
			private final Map<Object, Boolean> visited = new HashMap<>();
			int depth = 0;

			@Override
			public boolean isElementVisible(Viewer viewer, Object element) {
				depth++;
				Boolean isMatch = visited.get(element);
				if (isMatch == null) {
					visited.put(element, false);
					// Put in surrogate to prevent stack-overflows
					isMatch = super.isElementVisible(viewer, element);
					visited.put(element, isMatch);
				}
				depth--;
				if (depth == 0) {
					visited.clear();
				}
				return isMatch;
			}
		}, true, true);
```

Reply:
Do you have an example for such a cycle so that I can also test this on my side?

Reply:
The TP that is active in that case is the following wild mixture (I haven't stripped it down yet, but I think it's also not that important for this issue):
```
<target includeMode="feature" name="featureBased">
	<locations>
		<location includeAllPlatforms =   "false"   includeConfigurePhase  = "true" includeMode="planner" includeSource="true" type="InstallableUnit">
			<repository location="https://download.eclipse.org/releases/2023-12/"/>
			<unit id="org.eclipse.pde.feature.group" />
			<unit id="org.eclipse.m2e.pde.feature.feature.group"/>
			<unit id="jakarta.annotation-api" version="0.0.0"/>
			<unit id="org.eclipse.ecf.filetransfer.ssl.feature.feature.group" version="[1.0.0,2.0.0)"/>
			<unit id="org.eclipse.ecf.filetransfer.httpclient5.feature.feature.group" version="[1.0.0,2.0.0)"/>
			<unit id="org.eclipse.rcp.feature.group"/>
			<unit id="org.eclipse.ecf.core.feature.feature.group" />
			<unit id="org.eclipse.ecf.filetransfer.ssl.feature.source.feature.group" version="0.0.0"/>
			<unit id="org.eclipse.ecf.core.feature.source.feature.group"/>
			<unit id="org.eclipse.acceleo.feature.group" version="3.7.14.202311201319"/>
			<unit id="org.eclipse.acceleo.source.feature.group" version="3.7.14.202311201319"/>
			<unit id="com.google.guava"/>
			<unit id="org.eclipse.rap.jface"/>
			<unit id="org.eclipse.sdk.feature.group"/>
			<unit id="org.eclipse.m2e.sdk.feature.feature.group"/>
			<!--unit id="org.eclipse.emf.rap.sdk.feature.group"/-->
		</location>
	</locations>
</target>
```

Reply:
For the sake of documentation: The StackOverflowError can be reproduced using the following example configuration.
![image](https://github.com/user-attachments/assets/088e2d8c-da5f-406c-892c-0723ace07feb)